R/Figures.R

```diff
@@ -1138,6 +1138,385 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
+#' Scatterplot[[2]]
+#' 
+#' # Create multiple scatterplots and view the first plot in list.
+#' # In this example, we will group by both TADA.ComparableDataIdentifier
+#' # and MonitoringLocationTypeName (e.g. stream, reservoir, canal, etc.) and
+#' # because which MonitoringLocationTypeName was not specified, only the top 
+#' # 4 MonitoringLocationTypeName by field counts in the original dataset provided
+#' # will be included for each scatterplot generated.
+#' # Load example dataset:
+#' data(Data_Nutrients_UT)
+#' Scatterplot_output <- TADA_MultiScatterplot(Data_Nutrients_UT, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationTypeName"))
+#' # This example generates 14 scatterplots
+#' Scatterplot_output[[5]]
+#' Scatterplot_output[[8]]
+#' Scatterplot_output[[10]]
+#'
+#' # Load example dataset:
+#' data(Data_6Tribes_5y_Harmonized)
+#' # Filter the example data so it includes only one TADA.ComparableDataIdentifier
+#' df <- dplyr::filter(Data_6Tribes_5y_Harmonized, TADA.ComparableDataIdentifier %in% c("TOTAL PHOSPHORUS, MIXED FORMS_UNFILTERED_AS P_MG/L"))
+#' # Creates a scatterplot of the three specified sites of interest in the same plot.
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"), groups = c("Upper Red Lake: West", "Upper Red Lake: West-Central","Upper Red Lake: East Central"))
+#' # If no groups are selected, return the 4 groups (by MonitoringLocationName) with the greatest number of field count results for that TADA.ComparableDataIdentifier
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"))
+#' # Both commands below will return a single scatterplot for just the single comparable data identifier. 
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "TADA.ComparableDataIdentifier"))
+#' TADA_MultiScatterplot(df)
+#' 
+
+TADA_MultiScatterplot <- function(.data, id_cols = c("TADA.ComparableDataIdentifier"), groups = NULL) {
+  # check .data is data.frame
+  TADA_CheckType(.data, "data.frame", "Input object")
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, id_cols)
+  
+  # check .data has required columns
+  reqcols <- c(
+    "TADA.ResultMeasureValue",
+    "TADA.ResultMeasure.MeasureUnitCode",
+    "ActivityStartDate"
+  )
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, reqcols)
+  
+  # if left blank, ensure comparable data identifier is in the id_cols vector
+  if (is.null(id_cols) | length(unique(id_cols)) == 1) {
+    id_cols <- c("TADA.ComparableDataIdentifier","TADA.ComparableDataIdentifier")
+  }
+  id_cols <- c("TADA.ComparableDataIdentifier", vecsets::vsetdiff(id_cols, "TADA.ComparableDataIdentifier"))
+  
+  # if id_cols are specified by user and the comparable data identifier is not selected.
+  if (!"TADA.ComparableDataIdentifier" %in% id_cols) {
+    stop("Note: TADA.ComparableDataIdentifier not found in id_cols argument.")
```

Comment:
Should this be a "stop" or a  "print"? If we don't want the function to run if TADA.ComparableIDataIdentifier isn't a group_id, then it can just be automatically included as a grouping column. If we do want users to have some flexibility, for example to use TADA.CharacteristicName, instead of TADA.ComparableDataIdentifier, then use "print" to print a warning message.

Reply:
I think changing this to print makes sense